### ISSUE TITLE
SC1102 via SC2035 false positive in error accumulator

### ISSUE THREAD

**Issue by brackendawson** [koalaman/shellcheck#2705]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1102 & SC2035 
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

command
errors=$((errors + ?))

exit "$errors"
```

#### Here's what shellcheck currently says:

```
In - line 4:
errors=$((errors + ?))
       ^-- SC1102 (error): Shells disambiguate $(( differently or not at all. For $(command substitution), add space after $( . For $((arithmetics)), fix parsing errors.
                   ^-- SC2035 (info): Use ./*glob* or -- *glob* so names with dashes won't become options.
```

#### Here's what I wanted or expected to see:

No errors.

#### Debug

I think SC2035 is mistaking `?` for a glob when it is only an arithmetic operator and subsequently causing SC1102 to fire.

The actual practice of accumulating status codes is probably bad, but for proper uses of the ? variable in arithmetic this should be remedied.
